src/table/components/HeadCellMenu.tsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import ClickAwayListener from '@mui/material/ClickAwayListener';
+import Grow from '@mui/material/Grow';
+import Paper from '@mui/material/Paper';
+import Popper from '@mui/material/Popper';
+import MenuList from '@mui/material/MenuList';
+import List from '@mui/material/List';
+import ListItem from '@mui/material/ListItem';
+import ListItemButton from '@mui/material/ListItemButton';
+import ListItemIcon from '@mui/material/ListItemIcon';
+import ListItemText from '@mui/material/ListItemText';
+import ArrowUpwardIcon from '@mui/icons-material/ArrowUpward';
+import ArrowDownwardIcon from '@mui/icons-material/ArrowDownward';
+import MoreHoriz from '@mui/icons-material/MoreHoriz';
+import { StyledMenuIconButton, StyledCellMenu } from '../styles';
+import { GeneratedStyling } from '../types';
+
+const ListItems = ({ children, itemTitle }: { children: any; itemTitle: string }) => {
+  return (
+    <ListItem disablePadding>
+      <ListItemButton>
+        <ListItemIcon>{children}</ListItemIcon>
+        <ListItemText primary={itemTitle} />
+      </ListItemButton>
+    </ListItem>
+  );
+};
+
+export default function HeadCellMenu({ headerStyle }: { headerStyle: GeneratedStyling }) {
+  const [open, setOpen] = React.useState(false);
+  const anchorRef = React.useRef<HTMLButtonElement>(null);
+  // It returns focus to the button when we transitioned from !open -> open
+  const isOpen = React.useRef(open);
+
+  const handleClickOpen = (event: any) => {
+    event.stopPropagation();
```

Comment:
I tried two couple of options but because of the on the way issues couldn't make them complete:
- Not using stopPtopagation but instead making a callback that confirms the menu is open so disabling sorting function, but because of some reasons it couldn't update the setState value at the right time.
- Using stopPropagation  but moving the ClickAwayEventListener to very higher level component to TableCell but then one of useRef hooks didn't get the right value and a console error came up that I couldn't resolve it.
Which do you think might make more sense to continue with?

Reply:
Well, the main problem is that we have a button (open menu) inside a different button (sort). This is bad design. Not sure how we solve that, because it depends on the a11y design for the header... What I would do for now is to try to treat the sort "button" like it has a hole in it, where the menu button is. In the click handler for the sorting, check if the event comes from the menu button, and then early return.

Reply:
ok I'll remove it but what should I put instead? just the text with the bigger size padding ?

Reply:
no you wont actually change the styling. You just treat it as a "hole" in the click handler.
if (event.target === button) return

Reply:
Aha you mean 'sort' button, sorry I misunderstood first, got it now.